packages/server/models/subscriber.js

```diff
@@ -0,0 +1,47 @@
+import * as Joi from '@hapi/joi';
+import { v1 as uuidv1 } from 'uuid';
+import { database } from '../services';
+
+const COLLECTION = 'subscribers';
+
+const SubscriberSchema = Joi.object({
+  _id: Joi.string().default(uuidv1()),
+  id: Joi.string().default(uuidv1()),
+  email: Joi.string()
+    .email()
+    .required(),
+  status: Joi.boolean().default(true)
+});
+
+export const add = async input => {
+  return new Promise(async resolve => {
+    const { error, value } = await SubscriberSchema.validate(input);
+    if (error) return resolve({ error, data: value });
+
+    const recordError = await database
+      .ref([COLLECTION, value._id].join('/'))
```

Comment:
Để validate unique đơn giản hơn thì thay value._id  = value.email 
lúc check validate database.ref('subscribers/:email') => uniq hay không

Reply:
cái id nó từ chối value có dấu chấm và vài ký tự đặc biết nữa. Nên em sẽ hash cái email bằng md5 hoặc thuật toán tương tự để làm uniq ạ.